core/src/main/java/org/apache/druid/java/util/common/io/smoosh/FileSmoosher.java

```diff
@@ -82,6 +84,10 @@
   private List<File> completedFiles = new ArrayList<>();
   // list of files in process writing content using delegated smooshedWriter.
   private List<File> filesInProcess = new ArrayList<>();
+  // delegated smooshedWriter creates a new temporary file per file added. we use a counter to name these temporary
+  // files, and map the file number to the file name so we don't have to escape the file names (e.g. names with '/')
+  private int delegateFileCounter = 0;
```

Comment:
While this is likely only ever used in a single-threaded context.  The extra overhead of an `AtomicLong` is basically nothing and it makes for a bit nicer guarantees, so I'd suggest using that instead.

Reply:
this class definitely isn't thread-safe, but i don't have a strong preference either way so went ahead and changed it